### PR TITLE
Future dates are invalid and infer 'YY dates in Birthdate

### PR DESCRIPTION
**Pull request by rkuykendall**
JIRA 1: https://thatsmighty.atlassian.net/browse/MTY-506

JIRA 2: https://thatsmighty.atlassian.net/browse/MTY-507

### CODE REVIEW

File: src/inputs/Birthdate.tsx [hunk 1]
```diff
@@ -21,12 +21,16 @@ interface IValueObject {
 }
 
 interface IInputConfig {
-  padStart: number;
   style: { width: string, marginRight?: string };
 }
 
-export function isValidDate (value: string) {
-  return !!value && value.length === '####-##-##'.length && moment(value).isValid();
+export function isValidBirthdate (value: string) {
```
[the remark this answers is not in the document]
Comment (rkuykendall) [reply src/inputs/Birthdate.tsx]: Hmmmmm... I have some competing priorities here. Right now, the utilities and renders for all 'types' are kept in the file, totally self-contained. A good example of this is `Rate`:

image

I do agree this file is quite large though, and that these utilities could be robustly tested outside the class. I could move them into the utilities folder, but right now that's _exclusively _ `fields-ant` utilities, not utilities for specific types. I could either leave these as named exports in this file, or maybe `inputs/utilities.ts`? I'd also be open to moving some of the most generic ones like `inferCentury` into `@mighty-justice/utils`.

File: src/inputs/Birthdate.tsx [hunk 2]
```diff
@@ -66,10 +67,40 @@ class Birthday extends Component<IInputProps> {
     return this.valueObject[field] || '';
   }
 
+  private static inferCentury (year: string) {
```
[the remark this answers is not in the document]
Comment (rkuykendall) [reply src/inputs/Birthdate.tsx]: No, dates should always be ISO strings. Numbers are a very messy way of dealing with data like that. Like we wouldn't want to use a number for SSN or phone.

File: src/inputs/Birthdate.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ronaldzuniga) [reply src/inputs/Birthdate.tsx]: Let's move all the utilities mentioned into `@mighty-justice/utils`, I think that's the best place for them to be reused and as you said robustly tested outside of a UI layer.

File: src/inputs/Birthdate.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (rkuykendall) [reply src/inputs/Birthdate.tsx]: Moving much of this out into utils.

**Pull request merged**